cmd/up.go

```diff
@@ -162,6 +165,20 @@ func Up() *cobra.Command {
 
 //RunUp starts the up sequence
 func RunUp(dev *model.Dev, autoDeploy, build, forcePull, resetSyncthing bool) error {
+
+	if dev.Namespace == "" {
+		_, _, namespace, err := k8Client.GetLocal()
+		if err != nil {
+			return err
+		}
+		dev.Namespace = namespace
```

Comment:
The namespace value might come from the namespace flag too.
`dev.Namespace` is initialized at this point:
    https://github.com/okteto/okteto/blob/67e1c712d3251d2397d93c1e11349c43392eed42/cmd/up.go#L232
The `RunUp` logic is a bit messy, we need to refactor it. But you could bring the `dev.Namespace` initialization logic here if needed

Reply:
Sorry, @pchico83  I'm a little bit confused because the referenced line is older than the current. Can you explain more?

Reply:
Sure. In the master branch, `dev.Namespace` is initialized from lines 220-230 in `cmd/up.go`.
I think you can move that logic here to have a single place where where `k8Client.GetLocal()` is called:
<img width="524" alt="Screenshot 2020-05-04 at 21 45 24" src="https://user-images.githubusercontent.com/7474696/81006955-a67d9f80-8e50-11ea-9348-a8a752b34a3a.png">

Reply:
Got it, thanks.
Will do so 👍🏻